src/main/java/competition/subsystems/turret/TurretSubsystem.java

```diff
@@ -0,0 +1,87 @@
+package competition.subsystems.turret;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+
+import competition.IdealElectricalContract;
+import xbot.common.command.BaseSubsystem;
+import xbot.common.controls.actuators.XCANTalon;
+import xbot.common.injection.wpi_factories.CommonLibFactory;
+import xbot.common.properties.DoubleProperty;
+import xbot.common.properties.PropertyFactory;
+
+@Singleton
+public class TurretSubsystem extends BaseSubsystem {
+    
+    public XCANTalon motor;
+    public double currentAngle;
+    final DoubleProperty maxAngleProp;
+    final DoubleProperty minAngleProp;
+    final DoubleProperty turnPowerProp;
+
+    @Inject
+    public TurretSubsystem(CommonLibFactory factory, PropertyFactory pf, IdealElectricalContract contract ) {
+        log.info("Creating TurretSubsystem");
+        pf.setPrefix(this);
+        
+        currentAngle = 0;
+        maxAngleProp = pf.createPersistentProperty("Max Angle", 180);
+        minAngleProp = pf.createPersistentProperty("Min Angle", -180);
+        turnPowerProp = pf.createPersistentProperty("Turn Speed", .03);
+
+        
+
+        if(contract.isTurretReady())
+        {
+            this.motor = factory.createCANTalon(contract.rotationMotor().channel);
+            motor.setInverted(contract.rotationMotor().inverted);
+        }
+    }
+
+    public void turnLeft()
+    {
+        motor.simpleSet(turnPowerProp.get());
+    }
+
+    public void turnRight()
+    {
+        motor.simpleSet(-turnPowerProp.get());
+    }
+
+    public void setPower(double power) {
+        if((power<0 && canTurnLeft())||(power>0 && canTurnRight()))
+        {
+            motor.simpleSet(power);
+        }
+        else if(power == 0)
```

Comment:
should this just be an else { power = 0 } ?

Reply:
no because it will allow it to set the power to turn right even if it extends its bounds

Reply:
As we discussed, since the current code won't set anything to the motor in that case it will dangerously keep using the last value that it had. So instead we need to make sure we explicitly call motor.simpleSet(0) if it's at it's bounds and trying to go past them